.github/workflows/publish-apple-darwin-wasm.yml

```diff
@@ -11,8 +20,36 @@ on:
   push:
 
 jobs:
+  build-barretenberg:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: cachix/install-nix-action@v20
+        with:
+          nix_path: nixpkgs=channel:nixos-22.11
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: cachix/cachix-action@v12
+        with:
+          name: barretenberg
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+
+      # Upload does not work with symlinks, using this workaround:
+      # https://github.com/actions/upload-artifact/issues/92#issuecomment-1080347032    
+      - name: Build barretenberg as libbarretenberg-wasm32
+        run: |
+          nix build "git+https://github.com/AztecProtocol/barretenberg#wasm32"

```

Comment:
This won't be pinned to the bb commit that noir is pinned to. We need to be using the lockfile that noir is

Reply:
which lockfile?

Reply:
https://github.com/noir-lang/noir/blob/master/flake.lock#L15-L17

Reply:
pinning done in nix-barretenberg.yml